File: NGL.DataAssemblies/Ngl.FreightMaster.Data/DTO/CarrierData/CarrierEDI.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Runtime.Serialization
Imports Ngl.Core.ChangeTracker

Namespace DataTransferObjects
    <DataContract()> _
    Public Class CarrierEDI
        Inherits DTOBaseClass


#Region " Data Members"
        Private _CarrierEDIControl As Integer = 0
        <DataMember()> _
        Public Property CarrierEDIControl() As Integer
            Get
                Return _CarrierEDIControl
            End Get
            Set(ByVal value As Integer)
                _CarrierEDIControl = value
            End Set
        End Property

        Private _CarrierEDICarrierControl As Integer = 0
        <DataMember()> _
        Public Property CarrierEDICarrierControl() As Integer
            Get
                Return _CarrierEDICarrierControl
            End Get
            Set(ByVal value As Integer)
                _CarrierEDICarrierControl = value
            End Set
        End Property

        Private _CarrierEDIXaction As String = ""
        <DataMember()> _
        Public Property CarrierEDIXaction() As String
            Get
                Return Left(_CarrierEDIXaction, 3)
            End Get
            Set(ByVal value As String)
                _CarrierEDIXaction = Left(value, 3)
            End Set
        End Property

        Private _CarrierEDIComment As String = ""
        <DataMember()> _
        Public Property CarrierEDIComment() As String
            Get
                Return Left(_CarrierEDIComment, 50)
            End Get
            Set(ByVal value As String)
                _CarrierEDIComment = Left(value, 50)
            End Set
        End Property

        Private _CarrierEDISecurityQual As String = ""
        <DataMember()> _
        Public Property CarrierEDISecurityQual() As String
            Get
                Return Left(_CarrierEDISecurityQual, 2)
            End Get
            Set(ByVal value As String)
                _CarrierEDISecurityQual = Left(value, 2)
            End Set
        End Property

        Private _CarrierEDISecurityCode As String = ""
        <DataMember()> _
        Public Property CarrierEDISecurityCode() As String
            Get
                Return Left(_CarrierEDISecurityCode, 10)
            End Get
            Set(ByVal value As String)
                _CarrierEDISecurityCode = Left(value, 10)
            End Set
        End Property

        Private _CarrierEDIPartnerQual As String = ""
        <DataMember()> _
        Public Property CarrierEDIPartnerQual() As String
            Get
                Return Left(_CarrierEDIPartnerQual, 2)
            End Get
            Set(ByVal value As String)
                _CarrierEDIPartnerQual = Left(value, 2)
            End Set
        End Property

        Private _CarrierEDIPartnerCode As String = ""
        <DataMember()> _
        Public Property CarrierEDIPartnerCode() As String
            Get
                Return Left(_CarrierEDIPartnerCode, 15)
            End Get
            Set(ByVal value As String)
                _CarrierEDIPartnerCode = Left(value, 15)
            End Set
        End Property

        Private _CarrierEDIISASequence As Integer = 0
        <DataMember()> _
        Public Property CarrierEDIISASequence() As Integer
            Get
                Return _CarrierEDIISASequence
            End Get
            Set(ByVal value As Integer)
                _CarrierEDIISASequence = value
            End Set
        End Property

        Private _CarrierEDIGSSequence As Integer = 0
        <DataMember()> _
        Public Property CarrierEDIGSSequence() As Integer
            Get
                Return _CarrierEDIGSSequence
            End Get
            Set(ByVal value As Integer)
                _CarrierEDIGSSequence = value
            End Set
        End Property

        Private _CarrierEDIEmailNotificationOn As Boolean = False
        <DataMember()> _
        Public Property CarrierEDIEmailNotificationOn() As Boolean
            Get
                Return _CarrierEDIEmailNotificationOn
            End Get
            Set(ByVal value As Boolean)
                _CarrierEDIEmailNotificationOn = value
            End Set
        End Property

        Private _CarrierEDIEmailAddress As String = ""
        <DataMember()> _
        Public Property CarrierEDIEmailAddress() As String
            Get
                Return Left(_CarrierEDIEmailAddress, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIEmailAddress = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIAcknowledgementRequested As Boolean = False
        <DataMember()> _
        Public Property CarrierEDIAcknowledgementRequested() As Boolean
            Get
                Return _CarrierEDIAcknowledgementRequested
            End Get
            Set(ByVal value As Boolean)
                _CarrierEDIAcknowledgementRequested = value
            End Set
        End Property

        Private _CarrierEDIAcceptOn997 As Boolean = False
        <DataMember()> _
        Public Property CarrierEDIAcceptOn997() As Boolean
            Get
                Return _CarrierEDIAcceptOn997
            End Get
            Set(ByVal value As Boolean)
                _CarrierEDIAcceptOn997 = value
            End Set
        End Property

        Private _CarrierEDITestCode As String = ""
        <DataMember()> _
        Public Property CarrierEDITestCode() As String
            Get
                Return Left(_CarrierEDITestCode, 1)
            End Get
            Set(ByVal value As String)
                _CarrierEDITestCode = Left(value, 1)
            End Set
        End Property

        Private _CarrierEDIUpdated As Byte()
        <DataMember()> _
        Public Property CarrierEDIUpdated() As Byte()
            Get
                Return _CarrierEDIUpdated
            End Get
            Set(ByVal value As Byte())
                _CarrierEDIUpdated = value
            End Set
        End Property

        Private _CarrierEDIInboundFolder As String = ""
        <DataMember()> _
        Public Property CarrierEDIInboundFolder() As String
            Get
                Return Left(_CarrierEDIInboundFolder, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIInboundFolder = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIBackupFolder As String = ""
        <DataMember()> _
        Public Property CarrierEDIBackupFolder() As String
            Get
                Return Left(_CarrierEDIBackupFolder, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIBackupFolder = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDILogFile As String = ""
        <DataMember()> _
        Public Property CarrierEDILogFile() As String
            Get
                Return Left(_CarrierEDILogFile, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDILogFile = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIStartTime As String = ""
        <DataMember()> _
        Public Property CarrierEDIStartTime() As String
            Get
                Return Left(_CarrierEDIStartTime, 20)
            End Get
            Set(ByVal value As String)
                _CarrierEDIStartTime = Left(value, 20)
            End Set
        End Property

        Private _CarrierEDIEndTime As String = ""
        <DataMember()> _
        Public Property CarrierEDIEndTime() As String
            Get
                Return Left(_CarrierEDIEndTime, 20)
            End Get
            Set(ByVal value As String)
                _CarrierEDIEndTime = Left(value, 20)
            End Set
        End Property

        Private _CarrierEDIDaysOfWeek As String = ""
        <DataMember()> _
        Public Property CarrierEDIDaysOfWeek() As String
            Get
                Return Left(_CarrierEDIDaysOfWeek, 50)
            End Get
            Set(ByVal value As String)
                _CarrierEDIDaysOfWeek = Left(value, 50)
            End Set
        End Property

        Private _CarrierEDISendMinutesOutbound As Integer = 0
        <DataMember()> _
        Public Property CarrierEDISendMinutesOutbound() As Integer
            Get
                Return _CarrierEDISendMinutesOutbound
            End Get
            Set(ByVal value As Integer)
                _CarrierEDISendMinutesOutbound = value
            End Set
        End Property

        Private _CarrierEDIFileNameBaseOutbound As String = ""
        <DataMember()> _
        Public Property CarrierEDIFileNameBaseOutbound() As String
            Get
                Return Left(_CarrierEDIFileNameBaseOutbound, 20)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFileNameBaseOutbound = Left(value, 20)
            End Set
        End Property

        Private _CarrierEDIFileNameBaseInbound As String = ""
        <DataMember()> _
        Public Property CarrierEDIFileNameBaseInbound() As String
            Get
                Return Left(_CarrierEDIFileNameBaseInbound, 20)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFileNameBaseInbound = Left(value, 20)
            End Set
        End Property

        Private _CarrierEDIWebServiceAuthKey As String = ""
        <DataMember()> _
        Public Property CarrierEDIWebServiceAuthKey() As String
            Get
                Return Left(_CarrierEDIWebServiceAuthKey, 50)
            End Get
            Set(ByVal value As String)
                _CarrierEDIWebServiceAuthKey = Left(value, 50)
            End Set
        End Property

        Private _CarrierEDINGLEDIInputWebURL As String = ""
        <DataMember()> _
        Public Property CarrierEDINGLEDIInputWebURL() As String
            Get
                Return Left(_CarrierEDINGLEDIInputWebURL, 100)
            End Get
            Set(ByVal value As String)
                _CarrierEDINGLEDIInputWebURL = Left(value, 100)
            End Set
        End Property

        Private _CarrierEDINGLEDI204OutputWebURL As String = ""
        <DataMember()> _
        Public Property CarrierEDINGLEDI204OutputWebURL() As String
            Get
                Return Left(_CarrierEDINGLEDI204OutputWebURL, 1000)
            End Get
            Set(ByVal value As String)
                _CarrierEDINGLEDI204OutputWebURL = Left(value, 1000)
            End Set
        End Property

        Private _CarrierEDIOutboundFolder As String = ""
        <DataMember()> _
        Public Property CarrierEDIOutboundFolder() As String
            Get
                Return Left(_CarrierEDIOutboundFolder, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIOutboundFolder = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDILastOutboundTransmission As System.Nullable(Of Date)
        <DataMember()> _
        Public Property CarrierEDILastOutboundTransmission() As System.Nullable(Of Date)
            Get
                Return _CarrierEDILastOutboundTransmission
            End Get
            Set(ByVal value As System.Nullable(Of Date))
                _CarrierEDILastOutboundTransmission = value
            End Set
        End Property

        Private _CarrierEDIFTPOutboundFolder As String = ""
        <DataMember()> _
        Public Property CarrierEDIFTPOutboundFolder() As String
            Get
                Return Left(_CarrierEDIFTPOutboundFolder, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFTPOutboundFolder = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIFTPBackupFolder As String = ""
        <DataMember()> _
        Public Property CarrierEDIFTPBackupFolder() As String
            Get
                Return Left(_CarrierEDIFTPBackupFolder, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFTPBackupFolder = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIFTPInboundFolder As String = ""
        <DataMember()> _
        Public Property CarrierEDIFTPInboundFolder() As String
            Get
                Return Left(_CarrierEDIFTPInboundFolder, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFTPInboundFolder = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIFTPServer As String = ""
        <DataMember()> _
        Public Property CarrierEDIFTPServer() As String
            Get
                Return Left(_CarrierEDIFTPServer, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFTPServer = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIFTPUserName As String = ""
        <DataMember()> _
        Public Property CarrierEDIFTPUserName() As String
            Get
                Return Left(_CarrierEDIFTPUserName, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFTPUserName = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIFTPPassword As String = ""
        <DataMember()> _
        Public Property CarrierEDIFTPPassword() As String
            Get
                Return Left(_CarrierEDIFTPPassword, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIFTPPassword = Left(value, 255)
            End Set
        End Property

        Private _CarrierEDIWebServiceURL As String = ""
        <DataMember()> _
        Public Property CarrierEDIWebServiceURL() As String
            Get
                Return Left(_CarrierEDIWebServiceURL, 255)
            End Get
            Set(ByVal value As String)
                _CarrierEDIWebServiceURL = Left(value, 255)
            End Set
        End Property
        Private _CarrierEDIModDate As System.Nullable(Of Date) 'added for CarrierEDI new changes ManoRama 19AUG2020'
        <DataMember()>
        Public Property CarrierEDIModDate() As System.Nullable(Of Date)
            Get
                Return _CarrierEDIModDate
            End Get
            Set(ByVal value As System.Nullable(Of Date))
                _CarrierEDIModDate = value
            End Set
        End Property

        Private _CarrierEDIModUser As String = "" 'added for CarrierEDI new changes ManoRama 19AUG2020'
        <DataMember()>
        Public Property CarrierEDIModUser() As String
            Get
                Return Left(_CarrierEDIModUser, 100)
            End Get
            Set(ByVal value As String)
                _CarrierEDIModUser = Left(value, 100)
            End Set
        End Property
#End Region

#Region " Public Methods"
        Public Overrides Function Clone() As DTOBaseClass
            Dim instance As New CarrierEDI
            instance = DirectCast(MemberwiseClone(), CarrierEDI)
            Return instance
        End Function

#End Region

    End Class
End Namespace

File: DTMS365-AppSource/DynamicsTMS365/Models/CarrierEDI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DynamicsTMS365.Models
{
    public class CarrierEDI
    {
        public Carrier Carrier { get; set; }
        public bool CarrierEDIAcceptOn997 { get; set; }
        public bool CarrierEDIAcknowledgementRequested { get; set; }
        public string CarrierEDIBackupFolder { get; set; }
        public int? CarrierEDICarrierControl { get; set; }
        public string CarrierEDIComment { get; set; }
        public int CarrierEDIControl { get; set; }
        public string CarrierEDIDaysOfWeek { get; set; }
        public string CarrierEDIEmailAddress { get; set; }
        public bool CarrierEDIEmailNotificationOn { get; set; }
        public string CarrierEDIEndTime { get; set; }
        public string CarrierEDIFileNameBaseInbound { get; set; }
        public string CarrierEDIFileNameBaseOutbound { get; set; }
        public string CarrierEDIFTPBackupFolder { get; set; }
        public string CarrierEDIFTPInboundFolder { get; set; }
        public string CarrierEDIFTPOutboundFolder { get; set; }
        public string CarrierEDIFTPPassword { get; set; }
        public string CarrierEDIFTPServer { get; set; }
        public string CarrierEDIFTPUserName { get; set; }
        public int CarrierEDIGSSequence { get; set; }
        public string CarrierEDIInboundFolder { get; set; }

        public string _CarrierEDIInboundFolder { get { if (string.IsNullOrEmpty(this.CarrierEDIInboundFolder)) { this.CarrierEDIInboundFolder = @"C:\Data\Inbound"; } return this.CarrierEDIInboundFolder; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIInboundFolder = @"C:\Data\Inbound"; } } }
        public string _CarrierEDIBackupFolder { get { if (string.IsNullOrEmpty(this.CarrierEDIBackupFolder)) { return this.CarrierEDIBackupFolder = @"C:\Data\Backup"; } return this.CarrierEDIBackupFolder; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIBackupFolder = @"C:\Data\Backup"; } } }
        public string _CarrierEDILogFile { get { if (string.IsNullOrEmpty(this.CarrierEDILogFile)) { this.CarrierEDILogFile = @"C:\Data\EDILogFile.txt"; } return CarrierEDILogFile; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDILogFile = @"C:\Data\EDILogFile.txt"; } } }
        public string _CarrierEDIStartTime { get { if (string.IsNullOrEmpty(this.CarrierEDIStartTime)) { this.CarrierEDIStartTime = "09:00 AM"; } return CarrierEDIStartTime; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIStartTime = "09:00 AM"; } } }
        public string _CarrierEDIEndTime { get { if (string.IsNullOrEmpty(this.CarrierEDIEndTime)) { this.CarrierEDIEndTime = "05:00 PM"; } return CarrierEDIEndTime; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIEndTime = "05:00 PM"; } } }
        public string _CarrierEDIDaysOfWeek { get { if (string.IsNullOrEmpty(this.CarrierEDIDaysOfWeek)) { this.CarrierEDIDaysOfWeek = "CarrierEDIBackupFolder"; } return CarrierEDIDaysOfWeek; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIDaysOfWeek = "CarrierEDIBackupFolder"; } } }
        public string _CarrierEDIFileNameBaseOutbound { get { if (string.IsNullOrEmpty(this.CarrierEDIFileNameBaseOutbound)) { this.CarrierEDIFileNameBaseOutbound = "204OUT"; } return CarrierEDIFileNameBaseOutbound; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIFileNameBaseOutbound = "204OUT"; } } }
        public string _CarrierEDIFileNameBaseInbound { get { if (string.IsNullOrEmpty(this.CarrierEDIFileNameBaseInbound)) { this.CarrierEDIFileNameBaseInbound = "EDIIn"; } return CarrierEDIFileNameBaseInbound; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIFileNameBaseInbound = "EDIIn"; } } }
        public string _CarrierEDIWebServiceAuthKey { get { if (string.IsNullOrEmpty(this.CarrierEDIWebServiceAuthKey)) { this.CarrierEDIWebServiceAuthKey = "ABCDE"; } return CarrierEDIWebServiceAuthKey; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIWebServiceAuthKey = "ABCDE"; } } }
        public string _CarrierEDINGLEDIInputWebURL { get { if (string.IsNullOrEmpty(this.CarrierEDINGLEDIInputWebURL)) { this.CarrierEDINGLEDIInputWebURL = "http://localhost:4479/EDIInput.asmx"; } return CarrierEDINGLEDIInputWebURL; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDINGLEDIInputWebURL = "http://localhost:4479/EDIInput.asmx"; } } }
        public string _CarrierEDINGLEDI204OutputWebURL { get { if (string.IsNullOrEmpty(this.CarrierEDINGLEDI204OutputWebURL)) { this.CarrierEDINGLEDI204OutputWebURL = "http://localhost:4479/EDI204Output.asmx"; } return CarrierEDINGLEDI204OutputWebURL; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDINGLEDI204OutputWebURL = "http://localhost:4479/EDI204Output.asmx"; } } }
        public string _CarrierEDIOutboundFolder { get { if (string.IsNullOrEmpty(this.CarrierEDIOutboundFolder)) { this.CarrierEDIOutboundFolder = @"C:\Data\Outbound"; } return CarrierEDIOutboundFolder; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIOutboundFolder = @"C:\Data\Outbound"; } } }
        public string _CarrierEDIWebServiceURL { get { if (string.IsNullOrEmpty(this.CarrierEDIWebServiceURL)) { this.CarrierEDIWebServiceURL = "http://localhost:4479/EDIInput.asmx"; } return CarrierEDIWebServiceURL; } set { if (string.IsNullOrEmpty(value)) { this.CarrierEDIWebServiceURL = "http://localhost:4479/EDIInput.asmx"; } } }
        public string _CarrierEDISecurityQual { get { if (string.IsNullOrEmpty(this.CarrierEDISecurityQual)) { this.CarrierEDISecurityQual = "00"; } return CarrierEDISecurityQual; } set {  } }
   
        public string _CarrierEDIPartnerQual { get { if (string.IsNullOrEmpty(this.CarrierEDIPartnerQual)) { this.CarrierEDIPartnerQual = "02"; } return CarrierEDIPartnerQual; } set {  } }
          
        public char _CarrierEDITestCode { get { if (this.CarrierEDITestCode.ToString()=="\0") {  this.CarrierEDITestCode = 'P'; }return  this.CarrierEDITestCode; } set {   } }         
                    
        public int _CarrierEDIISASequence{ get{ if(this.CarrierEDIISASequence==0) this.CarrierEDIISASequence = 11111; return this.CarrierEDIISASequence; } set{  } }
        public int _CarrierEDISendMinutesOutbound { get { if (this.CarrierEDISendMinutesOutbound == 0) this.CarrierEDISendMinutesOutbound = 15; return CarrierEDISendMinutesOutbound; } set {  } }
        public int _CarrierEDIGSSequence { get { if (this.CarrierEDIGSSequence == 0) this.CarrierEDIGSSequence = 11111; return this.CarrierEDIGSSequence; } set { } } 
        
        public int CarrierEDIISASequence { get; set; }
        public DateTime? CarrierEDILastOutboundTransmission { get; set; }
        public string CarrierEDILogFile { get; set; }
        public string CarrierEDINGLEDI204OutputWebURL { get; set; }
        public string CarrierEDINGLEDIInputWebURL { get; set; }
        public string CarrierEDIOutboundFolder { get; set; }
        public string CarrierEDIPartnerCode { get; set; }
        public string CarrierEDIPartnerQual { get; set; }
        public string CarrierEDISecurityCode { get; set; }
        public string CarrierEDISecurityQual { get; set; }
        public int CarrierEDISendMinutesOutbound { get; set; }
        public string CarrierEDIStartTime { get; set; }
        public char CarrierEDITestCode { get; set; }
        public byte[] _CarrierEDIUpdated { get; set; }
        public string CarrierEDIWebServiceAuthKey { get; set; }
        public string CarrierEDIWebServiceURL { get; set; }
        public string CarrierEDIXaction { get; set; }

        public string CarrierEDIUpdated
        {
            get
            {
                if (this._CarrierEDIUpdated != null)
                {

                    return Convert.ToBase64String(this._CarrierEDIUpdated);

                }
                return string.Empty;
            }
            set
            {
                if (string.IsNullOrEmpty(value))
                {

                    this._CarrierEDIUpdated = null;

                }

                else
                {

                    this._CarrierEDIUpdated = Convert.FromBase64String(value);

                }

            }
        }

        public void setUpdated(byte[] val) { _CarrierEDIUpdated = val; }
        public byte[] getUpdated() { return _CarrierEDIUpdated; }
    }
}